datalad/cmd.py

```diff
@@ -207,21 +206,20 @@ def run(self, cmd, log_stdout=True, log_stderr=True,
 
             if status not in [0, None]:
                 msg = "Failed to run %r%s. Exit code=%d" \
-                      % (cmd, " under %r" % cwd if cwd else '', status)
+                    % (cmd, " under %r" % cwd if cwd else '', status)
                 lgr.error(msg)
-                raise RuntimeError(msg)
-
+                raise CommandError(str(cmd), msg, status, out[0], out[1])
             else:
                 self.log("Finished running %r with status %s" % (cmd, status), level=8)
 
         else:
             self.commands.append(cmd)
-            status, out = None, ("DRY", "DRY")
+            out = ("DRY", "DRY")
 
         if return_output:
-            return status, out
+            return out
         else:
-            return status
+            return None
```

Comment:
Let's completely remove return_output and always return it?
There will be no harm if not picked up outside ;)  would simplify interface